utils/land-patterns.stanza

```diff
@@ -1071,7 +1091,156 @@ defn ipc-formula (L:Toleranced, ; the distance from edge-of-lead to edge-of-lead
   val Gmin = Smax - 2.0 * Jh - C_S ; the distance from edge of land to edge of land on the interior of the land pattern
   val Xmin = Wmin + 2.0 * Js + C_W ; the size of the land in the dimension orthogonal to Z and G.
   IpcResults(Zmax, Gmin, Xmin)
-   
+
+; helper struct to contain information about how we plan 
+; to layout the pads on a single side of a quad land pattern
+public defn make-quad-landpattern (num-pins-west: Int,  ; the number of pins on the west face
+                                   num-pins-south: Int, ; the number of pins on the south face
+                                   num-pins-east: Int,  ; the number of pins on the east face
+                                   num-pins-north: Int, ; the number of pins on the north face
+                                   pitch-west:Double,   ; the pithc on the west face
+                                   pitch-south:Double,  ; the pitch on the south face
+                                   pitch-east:Double,   ; the pitch on the east face
+                                   pitch-north:Double,  ; the pitch on the north face
+                                   lead-span-1: Toleranced, ; the distance from pad-edge to pad-edge on the exterior of the part in the x direction
+                                   lead-span-2: Toleranced, ; the distance from pad-edge to pad-edge on the exterior of the part in the y direction
+                                   package-length: Toleranced,  ; the size of the package in the x direction
+                                   package-width: Toleranced,  ; the size of hte package in the y direction
+                                   lead-length: Toleranced, ; the dimension of the leads in the x direction on the east/west faces
+                                   lead-width: Toleranced, ; the dimension of the leads in the y direction on the east/west faces
+                                   lead-type:  LandProtrusionType, ; the type of the leads
+                                   density-level: DensityLevel   ; the density level of the design
+                                   ):
+  val lead-fillets = lead-fillets(lead-type, density-level)  
```

Comment:
`fatal("make-quad-landpattern(...) failed: pitch-west must be greater than 0. Called with: %_." % [pitch-west]) when pitch-west <= 0.0`

Reply:
I actually expect this to be less useful as an error message since the common case is all pitches being shared

Reply:
why 4 parameters if they are shared?

Reply:
They may be different, but I don't expect that to be the common case.